dynamic_analysis_pkg/dynamic_analysis_pkg/__init__.py

```diff
@@ -0,0 +1,167 @@
+'''Machinary to fetch simulation results to a Jupyter Notebook'''
+from pathlib import Path, PurePath
+from shutil import move
+from datetime import datetime
+import logging
+import os
+import pandas as pd
+from tabulate import tabulate
+import pyunicore.client as unicore_client
+from tqdm.notebook import tqdm
+
+L = logging.getLogger('dynamic-analysis-pkg')
+L.setLevel(logging.INFO)
+
+UNICORE_ENDPOINT = 'https://bspsa.cineca.it/advanced/pizdaint/rest/core'
+
+
+class AResults:
+    '''
+    Based on a simulation id, fetch the results using Unicore API
+    '''
+    DEFAULT_CIRCUIT_DIR = '/home/data-bbp/20191017/'
+    DEFAULT_WD_BASE = '/home/simulation-results'
+    ORIGIN_CIRCUIT_PATH = '/store/hbp/ich002/antonel/O1/20191017/'
+    HPC_DIR_BASE = '/scratch/snx3000/unicore/FILESPACE/'
+    JOB_BASE = os.path.join(UNICORE_ENDPOINT, 'jobs')
+
+    def __init__(self, token, sim_id, base_working_directory=None, new_circuit_path=None):
+        self.hpc_dir_path = None
+        self.files_list = None
+        self.circuit_dir = None
+        self.local_dir = None
+
+        if base_working_directory is None:
+            base_working_directory = self.DEFAULT_WD_BASE
+        if new_circuit_path is None:
+            new_circuit_path = self.DEFAULT_CIRCUIT_DIR
+
+        self.define_paths(sim_id, base_working_directory, new_circuit_path)
+        self.fetch_results(token, sim_id)
+        self.blueconfig = Path(self.local_dir) / 'BlueConfig'
+
+    def fetch_results(self, token, sim_id=None):
+        if not sim_id:
+            raise ValueError('[Error] Simulation ID not provided')
+
+        self.retrieve_sim_info(token, sim_id)
+        self.get_sim_results()
+
+    def retrieve_sim_info(self, token, sim_id=None):
+        tr = unicore_client.Transport(token)
+        job_url = os.path.join(self.JOB_BASE, sim_id)
+        job = unicore_client.Job(tr, job_url)
+        job_name = job.properties['name']
+        L.info('Fetching results of [%s]. Please wait ...', job_name)
+        storage = job.working_dir
+        self.files_list = storage.listdir()
+
+    def get_sim_results(self):
+        self.download_file_to_storage('BlueConfig')
+        self.download_report()
+        self.download_file_to_storage('out.dat')
+        L.info('Result were saved at: %s', self.local_dir)
+
+    def define_paths(self, sim_id, working_directory, new_circuit_path):
+        if not Path(new_circuit_path).is_dir():
+            raise ValueError('[ERROR] Circuit path does not exists')
+        else:
+            self.circuit_dir = new_circuit_path
+
+        self.local_dir = Path(working_directory) / Path(sim_id)
+        self.local_dir.mkdir(parents=True, exist_ok=True)
+
+        self.hpc_dir_path = Path(self.HPC_DIR_BASE) / Path(sim_id)
+
+    def download_file_to_storage(self, file_name):
+        ''' function to download the file and add it to kernel file system '''
+        file_output_path = Path(self.local_dir) / Path(file_name)
```

Comment:
The only place where you should need the `Path` constructor is when you define the paths.
On this line for examples, `self.local_dir` is already a `Path` object so you can simply:
```suggestion
        file_output_path = self.local_dir / file_name
```